src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2752,6 +2752,32 @@ private unsafe void SendManifest(byte[]? rawManifest)
 #endif // FEATURE_MANAGED_ETW
         }
 
+        // Helper to deal with the fact that the type we are reflecting over might be loaded in the ReflectionOnly context.
+        // When that is the case, we have the build the custom assemblies on a member by hand.
+        internal static bool IsCustomAttributeDefinedHelper(
+            MemberInfo member,
+            Type attributeType,
+            EventManifestOptions flags = EventManifestOptions.None)
+        {
+            // AllowEventSourceOverride is an option that allows either Microsoft.Diagnostics.Tracing or
+            // System.Diagnostics.Tracing EventSource to be considered valid.  This should not mattter anywhere but in Microsoft.Diagnostics.Tracing (nuget package).
+            if (!member.Module.Assembly.ReflectionOnly && (flags & EventManifestOptions.AllowEventSourceOverride) == 0)
+            {
+                // Let the runtime to the work for us, since we can execute code in this context.
+                return member.IsDefined(attributeType);
```

Comment:
```suggestion
                return member.IsDefined(attributeType, inherit: false);
```

The default `inherit` for `member.IsDefined` is `true`. But the `GetCustomAttributeHelper` method uses `false` below.

https://github.com/dotnet/runtime/blob/8636dc8a76d687169db64033b71ec9f79d9ba4b5/src/coreclr/System.Private.CoreLib/src/System/Attribute.CoreCLR.cs#L475-L478